### ISSUE TITLE
Fix `no-template-arrow` false positives

### ISSUE THREAD

**Issue by jpzwarte** [43081j/eslint-plugin-lit#189]
There are currently marked as invalid:
- `@click=${() => this.#onRetry(file)}`
- `@click=${async () => await this.#onDelete(file)}`

I believe they are false positives since the arguments to the method do not match the arguments of the arrow function. 

The rule should only be applicable when the arguments to the arrow function *match* the arguments of the called function.

**Comment by jpzwarte** [comment]
> in your situation, i'd just turn it off since your stylistic preference doesn't agree with it. and **maybe remove it from the recommended rule set**.

Right, I've already disabled it. And to avoid future confusion, I think removing it from the recommended rule set might be a good idea!
 
> if we repurpose the rule to be about _types_ (e.g. args matching), we're stepping on the toes of typescript and probably shouldn't be.

Right, as you explained, this would change the intention of the rule significantly.

**Comment by 43081j** [comment]
ill open a pr when i get chance to remove it from the recommended config (unless you beat me to it!)

thanks for raising it, its a pretty old rule and doesn't have as much meaning anymore
